src/otx/core/data/dataset/classification.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTXClassificationDataset."""
+
+from __future__ import annotations
+
+from typing import Callable, Optional
+
+import cv2
+import torch
+from datumaro import DatasetSubset, Image, Label
+
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.entity.classification import MulticlassClsBatchDataEntity, MulticlassClsDataEntity
+
+from .base import OTXDataset, Transforms
+
+
+class OTXMulticlassClsDataset(OTXDataset[MulticlassClsDataEntity]):
+    """OTXDataset class for multi-class classification task."""
+
+    def __init__(self, dm_subset: DatasetSubset, transforms: Transforms) -> None:
+        super().__init__(dm_subset, transforms)
+
+    def _get_item_impl(self, index: int) -> Optional[MulticlassClsDataEntity]:
+        item = self.dm_subset.get(id=self.ids[index], subset=self.dm_subset.name)
+
+        img = item.media_as(Image)
+        img_data = img.data
+        # TODO(vinnamkim): This is a temporal approach
+        # There is an upcoming Datumaro patch here for this
+        # https://github.com/openvinotoolkit/datumaro/pull/1194
+        if img_data.shape[-1] == 4:
+            img_data = cv2.cvtColor(img_data, cv2.COLOR_BGRA2BGR)
+        img_shape = img.size
```

Comment:
Would you please wrap up this with `OTXDetectionDataset` into the common function and add a branch for grayscale image (2D tensor)?

Reply:
Done by https://github.com/openvinotoolkit/training_extensions/pull/2678/commits/a806bce3701546431096b45507fc968cacc2d577 https://github.com/openvinotoolkit/training_extensions/pull/2678/commits/91cbe85b1cb1cecc0a6633b6d44da0217694de4a